packages/core/src/context/hotkeys/hotkeys-provider.md

```diff
@@ -43,6 +43,74 @@ ReactDOM.render(
 );
 ```
 
+@## Advanced usage
+
+HotkeysProvider should not be nested, except in special cases. If you have a rendering boundary within your application
+through which React context is not preserved (for example, a plugin system which uses `ReactDOM.render()`) and you wish
+to use hotkeys in a descendant part of the tree below such a boundary, you may render a descendant provider and hydrate it
+with the root context instance. Without this context plumbing, you would end up with two "global" hotkeys dialogs in an
```

Comment:
paragraph structure is slightly awkward. this presents a potential problem after the solution. maybe we should just clearly state the intended outcome, which is a single global hotkeys dialog no matter how many HotkeysProviders are rendered in the application.